src/api/app/jobs/send_event_emails.rb

```diff
@@ -33,4 +40,13 @@ def create_rss_notifications(event)
       )
     end
   end
+
+  def needs_logging?(event)
```

Comment:
What about having this as a method called **logged?** or something like that in the **Event** class returning false and override it at **Event:Project** and **Event::Package** classes returning **!project_logged**, that way you don't have to calculate if the class of the event is a descendant in this condition, it looks too strange here to me.